### PR TITLE
Do not place windows off screen

### PR DESCRIPTION
**Pull request by jnugh**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

This fixes an issue where a window could be placed off screen which happend when the display it was previously on has been disconnected before launching the application.

> The application will always be placed on an existing screen.

https://github.com/mattermost/desktop/issues/378

**Test Cases**
* Have 2 monitors
* Place the app on one of them
* Close the app
* Start the app -> It should be exactly were it was when you closed it
* Close the app
* Disable the monitor
* Start the app -> It should be on your main screen

**Additional Notes**
I do not check if the window fits onto the screen , so if you plug in another monitor with a smaller resolution the window might not fit the screen (which I actually didn't test, might as well be covered by electron).

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @jnugh!! 

I don't have two monitors to test..so wondering if someone in the community would like to help with testing?

[You can find the test builds here](https://circleci.com/gh/mattermost/desktop/809#artifacts), and test cases have been included above.

If you have two monitors with different sizes, please help test the corner case provided by @jnugh above.

**Comment by yuya-oc** [comment]
I could not reproduce the issue on Ubuntu 16.04 hosted on VirtualBox. However this feature looks like working well.

**Comment by jasonblais** [comment]
Let me see if I can find someone who can help test  this PR on Mac/Windows.

File: src/main.js [hunk 1]
```diff
@@ -201,6 +201,31 @@ function clearAppCache() {
   }
 }
 
+function restoreWindowState(state) {
```
Comment (jnugh) [inline src/main.js]: validateWindowPosition sounds as if it would return a bool value, I will go for getValidWindowPosition

**Comment by jasonblais** [comment]
@yuya-oc would you be able to test this on Windows, just to confirm it's working as expected?

I have a couple of people I can ask to try as well, but it would be good to try ourselves first (I would, but again only have one monitor).

**Comment by yuya-oc** [comment]
As I had written in https://github.com/mattermost/desktop/issues/378#issuecomment-267035909, the original issue didn't reproduced on my MacBook. But this feature works well if I launched the app from CLI.

For Windows, I need to bring another monitor to test this, so I will check this in weekend.

**Comment by yuya-oc** [comment]
Tested on Windows 10. In the case of that the secondary monitor is located at left of main monitor, `x` has a negative value. So the feature doesn't work.

secondary_is_located_at_left

File: src/main.js [hunk 2]
```diff
@@ -201,6 +201,31 @@ function clearAppCache() {
...
+  const displays = screen.getAllDisplays();
+  let maxX = 0;
+  let maxY = 0;
+  for (let i = 0; i < displays.length; i++) {
+    const display = displays[i];
+    maxX = Math.max(maxX, display.bounds.x + display.bounds.width);
+    maxY = Math.max(maxY, display.bounds.y + display.bounds.height);
+  }
+
+  if (state.x > maxX || state.y > maxY) {
```
Comment (yuya-oc) [inline src/main.js]: `(state.x < minX || state.y < minY)` should also be handled.

**Comment by jnugh** [comment]
You are absolutely right, could you please check the updated version.

**Comment by yuya-oc** [comment]
Tested on Windows 10 by editing `<userData>/bound-info.json`. The feature works well.
